src/navigation/RegisterENSNavigator.js

```diff
@@ -1,67 +1,142 @@
 import { createMaterialTopTabNavigator } from '@react-navigation/material-top-tabs';
-import React, { useEffect } from 'react';
+import ConditionalWrap from 'conditional-wrap';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 import { StatusBar } from 'react-native';
 import { SheetHandleFixedToTopHeight, SlackSheet } from '../components/sheet';
-import ENSAssignRecordsSheet from '../screens/ENSAssignRecordsSheet';
+import ENSAssignRecordsSheet, {
+  ENSAssignRecordsBottomActions,
+} from '../screens/ENSAssignRecordsSheet';
 import ENSSearchSheet from '../screens/ENSSearchSheet';
 import ScrollPagerWrapper from './ScrollPagerWrapper';
 import { sharedCoolModalTopOffset } from './config';
 import { Box } from '@rainbow-me/design-system';
-import { useDimensions } from '@rainbow-me/hooks';
+import { useDimensions, useKeyboardHeight } from '@rainbow-me/hooks';
 import Routes from '@rainbow-me/routes';
 import { deviceUtils } from '@rainbow-me/utils';
 
 const Swipe = createMaterialTopTabNavigator();
 
 const renderTabBar = () => null;
 const renderPager = props => (
-  <ScrollPagerWrapper {...props} initialScrollPosition={1} />
+  <ScrollPagerWrapper
+    {...props}
+    initialScrollPosition={1}
+    useViewPagerAdaptor={false}
+  />
 );
 
+const defaultScreenOptions = {
+  [Routes.ENS_ASSIGN_RECORDS_SHEET]: {
+    scrollEnabled: true,
+    useAccentAsSheetBackground: true,
+  },
+  [Routes.ENS_SEARCH_SHEET]: {
+    scrollEnabled: false,
+    useAccentAsSheetBackground: false,
+  },
+};
+
+const initialRouteName = Routes.ENS_SEARCH_SHEET;
+
 export default function RegisterENSNavigator() {
+  const sheetRef = useRef();
+
   const { height: deviceHeight } = useDimensions();
 
-  const extraTopPadding = android ? 29 : 19;
-  const contentHeight =
-    deviceHeight -
-    SheetHandleFixedToTopHeight -
-    sharedCoolModalTopOffset -
-    extraTopPadding;
+  const [currentRouteName, setCurrentRouteName] = React.useState(
+    initialRouteName
+  );
+
+  const screenOptions = useMemo(() => defaultScreenOptions[currentRouteName], [
+    currentRouteName,
+  ]);
+
+  const [scrollEnabled, setScrollEnabled] = useState(
+    screenOptions.scrollEnabled
+  );
+  useEffect(() => {
+    // Wait 200ms to prevent transition lag
+    setTimeout(() => {
+      setScrollEnabled(screenOptions.scrollEnabled);
+    }, 200);
+  }, [screenOptions.scrollEnabled]);
 
   useEffect(() => {
     StatusBar.setBarStyle('light-content');
   }, []);
 
+  useEffect(() => {
+    if (!screenOptions.scrollEnabled) {
+      sheetRef.current.scrollTo({ animated: false, x: 0, y: 0 });
+    }
+  }, [screenOptions.scrollEnabled]);
+
+  const keyboardHeight = useKeyboardHeight();
+  const bottomActionHeight = useMemo(() => (ios ? keyboardHeight : 250), [
+    keyboardHeight,
+  ]);
+  const contentHeight =
+    deviceHeight - SheetHandleFixedToTopHeight - sharedCoolModalTopOffset;
+
+  const isBottomActionsVisible =
```

Comment:
can we move these const to the top of the component just to have it ordered as other components in the app?